webapp/tests/elster_client/test_elster_client.py

```diff
@@ -106,6 +107,28 @@ def test_if_eric_transfer_error_occurred_raise_error(self):
         finally:
             MockErica.eric_transfer_error_occurred = False
 
+    def test_if_erica_timeout_error_occurred_raise_error(self):
+        MockErica.invalid_request_timeout_occurred = True
+        try:
+            with patch('requests.post', side_effect=MockErica.mocked_elster_requests), \
+                    patch('app.elster_client.elster_client.current_user',
+                          MagicMock(is_active=True, is_authenticated=True)):
+                self.assertRaises(EricaRequestTimeoutError, send_est_with_elster, self.valid_form_data, 'IP',
+                                  include_elster_responses=False)
+        finally:
+            MockErica.invalid_request_timeout_occurred = False
+
+    def test_if_erica_connection_error_occurred_raise_error(self):
+        MockErica.invalid_request_connection_error_occurred = True
+        try:
+            with patch('requests.post', side_effect=MockErica.mocked_elster_requests), \
+                    patch('app.elster_client.elster_client.current_user',
+                          MagicMock(is_active=True, is_authenticated=True)):
+                self.assertRaises(EricaRequestConnectionError, send_est_with_elster, self.valid_form_data, 'IP',
+                                  include_elster_responses=False)
+        finally:
+            MockErica.invalid_request_connection_error_occurred = False
+
```

Comment:
Nice.
I would have assumed that we test that on the level of the send_to_erica and directly setting the timeout error. Something like that for all three tests:
```python
def test_if_erica_timeout_error_occurred_then_raise_corresponding_error(self):
        old_use_mock_api_value = Config.USE_MOCK_API
        Config.USE_MOCK_API = False
        try:
            with patch('requests.post', side_effect=requests.Timeout(response=MockResponse(None, '503'))):
                with pytest.raises(EricaRequestTimeoutError):
                    send_to_erica()
        finally:
            Config.USE_MOCK_API = old_use_mock_api_value
```
Otherwise we would only test it for the `send_est_to_erica` but not for all the other requests we send to erica. Does this make sense to you?
The other tests for Erica test the result of the call to erica and the errors are raise in the corresponding send_est_to_erica function.

Reply:
I thought testing in one of the request methods was sufficient, but so it's less code and generally tested.

Reply:
:heavy_check_mark: